mpcs/src/basefold/query_phase.rs

```diff
@@ -157,239 +134,172 @@ pub fn batch_verifier_query_phase<E: ExtensionField, Spec: BasefoldSpec<E>>(
     let mmcs_ext = ExtensionMmcs::<E::BaseField, E, _>::new(poseidon2_merkle_tree::<E>());
     let mmcs = poseidon2_merkle_tree::<E>();
     let check_queries_span = entered_span!("check_queries");
-    // can't use witin_comm.log2_max_codeword_size since it's untrusted
-    let log2_witin_max_codeword_size =
-        max_num_var + <Spec::EncodingScheme as EncodingScheme<E>>::get_rate_log();
+    let log2_blowup = S::get_rate_log();
+    let log2_max_codeword_size = max_num_var + log2_blowup;
 
-    // an vector with same length as circuit_meta, which is sorted by num_var in descending order and keep its index
-    // for reverse lookup when retrieving next base codeword to involve into batching
-    let folding_sorted_order = circuit_meta
+    indices
         .iter()
-        .enumerate()
-        .sorted_by_key(|(_, CircuitIndexMeta { witin_num_vars, .. })| Reverse(witin_num_vars))
-        .map(|(index, CircuitIndexMeta { witin_num_vars, .. })| (witin_num_vars, index))
-        .collect_vec();
+        .zip_eq(proof.query_opening_proof.iter())
+        .for_each(
+            |(
+                idx,
+                QueryOpeningProof {
+                    input_proofs,
+                    commit_phase_openings: opening_ext,
+                },
+            )| {
+                // verify base oracle query proof
+                // refer to prover documentation for the reason of right shift by 1
+                let mut idx = idx >> 1;
 
-    indices.iter().zip_eq(queries).for_each(
-        |(
-            idx,
-            QueryOpeningProof {
-                witin_base_proof:
-                    BatchOpening {
-                        opened_values: witin_opened_values,
-                        opening_proof: witin_opening_proof,
-                    },
-                fixed_base_proof: fixed_commit_option,
-                commit_phase_openings: opening_ext,
-            },
-        )| {
-            // verify base oracle query proof
-            // refer to prover documentation for the reason of right shift by 1
-            let mut idx = idx >> 1;
+                let mut reduced_openings = BTreeMap::new();
+                let mut batch_coeffs_iter = batch_coeffs.iter();
 
-            let (witin_dimentions, fixed_dimentions) =
-                get_base_codeword_dimentions::<E, Spec>(circuit_meta);
-            // verify witness
-            mmcs.verify_batch(
-                &witin_comm.commit,
-                &witin_dimentions,
-                idx,
-                witin_opened_values,
-                witin_opening_proof,
-            )
-            .expect("verify witin commit batch failed");
+                for ((commit, batch_opening), input_proof) in
+                    rounds.iter().zip_eq(input_proofs.iter())
+                {
+                    let dimensions = batch_opening
+                        .iter()
+                        .filter(|(num_var, _)| *num_var >= S::get_basecode_msg_size_log())
```

Comment:
should we remove this filter? as batch_opening shouldn't have `num_var >= S::get_basecode_msg_size_log()`